db/prompts.go

```diff
@@ -158,11 +148,56 @@ var AllowedColumns = map[string]bool{
 }
 
 var usesField = `array(SELECT prompts_uses.created_at FROM prompts_uses WHERE prompts_uses.prompt_id = prompts.id ) as uses`
-var minField = `prompts.id, prompts.name, prompts.description, prompts.use, prompts.tags, prompts.public, prompts.user_id`
-var maxField = `prompts.id, prompts.name, prompts.description, prompts.prompt, prompts.created_at, prompts.updated_at, prompts.tags, prompts.public, prompts.user_id`
+var likesField = `(SELECT COUNT(*) FROM prompts_likes WHERE prompts_likes.prompt_id = prompts.id) as likes`
+
+var minField = `prompts.id, prompts.name, prompts.description, prompts.use, prompts.tags, prompts.public, prompts.user_id, prompts.slug`
+var maxField = `prompts.id, prompts.name, prompts.description, prompts.prompt, prompts.created_at, prompts.updated_at, prompts.tags, prompts.public, prompts.user_id, prompts.slug`
+
+func GetPromptByIdOrSlug(id string) (*Prompt, error) {
+	fieldMappings := map[string]string{
+		"uuid": "id",
+		"slug": "slug",
+	}
+
+	query, err := PrepareQueryByStringIdentifier(id, fieldMappings)
+	if err != nil {
+		return nil, err
+	}
+
+	prompt := &Prompt{}
+
+	result := DB.First(prompt, query, id)
+
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return prompt, nil
+}
+
+func getIsLikeField(userId string) string {
+	if userId == "" {
+		return ""
+	}
+	return fmt.Sprintf(`EXISTS (SELECT 1 FROM prompts_likes WHERE prompts_likes.prompt_id = prompts.id AND prompts_likes.user_id = '%s') as is_liked`, userId)
```

Comment:
The userId is not checked here. I don't think you can inject something in that particular case bc I guess the userId comes from a server generated token, but again it means the function getIsLikeField cannot be trusted. A string added to a raw SQL query should ALWAYS be checked right before adding it. If we don't we will forget at some point to check it before using the function it's in.